src/Handler/StreamHandler.php

```diff
@@ -317,27 +313,27 @@ private function createStream(RequestInterface $request, array $options)
         }
 
         // Microsoft NTLM authentication only supported with curl handler
-        if (isset($options['auth'])
-            && \is_array($options['auth'])
-            && isset($options['auth'][2])
-            && 'ntlm' == $options['auth'][2]
-        ) {
+        if (isset($options['auth'][2]) && 'ntlm' === $options['auth'][2]) {
             throw new \InvalidArgumentException('Microsoft NTLM authentication only supported with curl handler');
         }
 
         $uri = $this->resolveHost($request, $options);
 
-        $context = $this->createResource(
-            function () use ($context, $params) {
+        $contextResource = $this->createResource(
+            static function () use ($context, $params) {
                 return \stream_context_create($context, $params);
             }
         );
 
         return $this->createResource(
-            function () use ($uri, &$http_response_header, $context, $options) {
-                $resource = \fopen((string) $uri, 'r', null, $context);
+            function () use ($uri, &$http_response_header, $contextResource, $context, $options, $request) {
+                $resource = \fopen((string) $uri, 'r', false, $contextResource);
                 $this->lastHeaders = $http_response_header;
 
+                if (false === $resource) {
+                    throw new ConnectException('Connection refused for URI ' . $uri, $request, null, $context);
```

Comment:
sprintf should be used here?

Reply:
Fixed in 9eb2f5ab6d17224c9b83c311a81a79311f847a10